### DISCUSSION TITLE
multiline regex matching

### DISCUSSION THREAD

**Discussion by skepley** [texstudio-org/texstudio#1900] [Q&A] [answered]
I am trying to match a regular expression over multiple lines in a macro and it doesn't seem to work. It also doesn't work in the find menu which makes me wonder if there  is something subtle about the regex implementation. I thought it was based on javascript. But then the following regex: 

ab([\s\S]*)c

should match the text:

ab
c

but it does not. However, if I remove the line break e.g.

ab     c

it matches without issues. Is there some special syntax I need to use for matching over multiple lines? Thanks.

**Comment by sunderme** [comment, marked as answer]
txs search only works per line.

**Reply by skepley** [reply]
Thanks for the reply. I will have to find some workaround.

**Reply by Explorer-cc** [reply]
@sunderme  Is there any workaround? Or why TXS only support search in line?
If I want to use `^\s*\r?\n` to elliminate empty line, which is a common requirement, but it doesn't work in TXS.

**Reply by sunderme** [reply]
To work on a complete file, the best solution is to use javascript macros, see https://github.com/texstudio-org/texstudio/wiki/Scripts and https://texstudio-org.github.io/advanced.html#script-macros

In your case, e.g. https://github.com/texstudio-org/texstudio/wiki/Scripts#remove-all-empty-lines

**Reply by Explorer-cc** [reply]
ops! I missed it in the manual, thanks!
